src/main/java/io/jenkins/plugins/appcenter/api/AppCenterServiceFactory.java

```diff
@@ -88,6 +89,56 @@ public UploadService createUploadService(@Nonnull final String uploadUrl) {
         return retrofit.create(UploadService.class);
     }
 
+    public BlobClient createBlobUploadService(@Nonnull final String uploadUrl) {
+        //Workaround for bug in Azure Blob Storage, as AppCenter returns the upload URL with a port attached
+        //See https://github.com/Azure/azure-sdk-for-java/issues/15827
+        final HttpUrl httpUploadUrl = HttpUrl.get(uploadUrl);
+        final String file;
+        if (httpUploadUrl.encodedQuery() != null) {
+            file = httpUploadUrl.encodedPath() + "?" + httpUploadUrl.encodedQuery();
+        } else {
+            file = httpUploadUrl.encodedPath();
+        }
+
+        final String uploadUrlWithoutPort = String.format("%1$s://%2$s%3$s", httpUploadUrl.scheme(), httpUploadUrl.host(), file);
+
+        // For the future it would be nice to simply configure this with the azure-core-http-okhttp artefact so that we can make use of the shared OkHttp configuration for
+        // timeouts and proxy settings. For the time being configure this manually. https://github.com/Azure/azure-sdk-for-java/wiki/HTTP-clients
+
+
+        final NettyAsyncHttpClientBuilder nettyAsyncHttpClientBuilder = new NettyAsyncHttpClientBuilder();
+
+        if (proxyConfiguration != null) {
```

Comment:
Apologies this was something that I forgot to check in the original review but we should be conscious that some users might have a proxy in between their Jenkins box and the internet. Fortunately we can configure one like this.